bin/k8s-deploy

```diff
@@ -152,6 +152,22 @@ do
     echo "Updating ${DEPLOYMENT_FILE}-${CI_SHA1}"
     DEPLOYMENT_FILE=${DEPLOYMENT_FILE}-${CI_SHA1}
   fi
+
+
+  # This is workaround attended to address an HPA issue where the correct number of pods are not deployed when
+  # 'kubectl apply' is used to redeploy a deployment that has been scaled by an HPA.  There is an open issue for
+  # this: https://github.com/kubernetes/kubernetes/issues/25238
+
+  # If we can match 'replicas: hpa' in te deployment file
+  if [[ -n $( awk -F: '/[rR][eE][pP][lL][iI][cC][aA][sS][[:space:]]*:[[:space:]]*+[hH][pP][aA]/' $DEPLOYMENT_FILE ) ]]
+  then
+    # Get the replica count and sed it into the deployment file in place of 'hpa'
+    REPLICA_COUNT=$(kubectl get deployment ${DEPLOYMENT} --namespace=${NAMESPACE} -o json | jq '.status.readyReplicas')
+    echo "Updating replica count to ${REPLICA_COUNT} for ${DEPLOYMENT_FILE}"
+    sed -e 's/replicas: hpa/replicas: '${REPLICA_COUNT}'/gI;' ${DEPLOYMENT_FILE} > ${DEPLOYMENT_FILE}-hpa
```

Comment:
Noticing that the `sed` line needs to match the regex used above. If we accept that `HPA` should be usable this `sed` would fail to replace as it looks for `hpa`. Maybe we should use a rok8s-scripts unique value like `ROK8S_HPA_REPLICAS` to make the replacement easier?

Reply:
The regex for sed is slightly different, sed is using the /gI pattern flag which will result in it ignoring case.

Reply:
The `I` handles the case sensitivity. There is a bit of a difference for how whitespace is handled but likely fine for now and see how this performs in the real world!